### PR TITLE
adds use-case for flagging bad-clients

### PR DESCRIPTION
**Pull request by sfrehse**
This PR is part of a technical article we are preparing.

Content: How to make bad client behavior more visual. The idea behind is to re-route bad client MQTT messages into a dedicated topic and use the postgres extension to store them in a database. With Grafana we visualize the top most 10 clients,

### CODE REVIEW

File: examples/flagging-bad-clients/hivemq/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM hivemq/hivemq4:4.16.0
+
+# postgre
+ADD extension/hivemq-postgresql-extension-0.1.0-RC-1-proguarded.zip /opt/hivemq/extensions
```
Comment (gitseti) [inline examples/flagging-bad-clients/hivemq/Dockerfile:4]: This extension is neither public nor checked out in this project.
So either we need to mention in the quickstart, that users need to download the extension themselves from somewhere (?) and put it into the extension folder, or include the pre-release zip in this project for now. 

Will the extension be publicly downloadable? Then we can do the same as with the prometheus extension in L.14 as soon as the extension is released.

File: examples/flagging-bad-clients/hivemq/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM hivemq/hivemq4:4.16.0
+
+# postgre
```
Comment (gitseti) [inline examples/flagging-bad-clients/hivemq/Dockerfile:3]:
```suggestion
# postgres extension
```

File: examples/flagging-bad-clients/hivemq/Dockerfile (hunk 1 above)
Comment (sfrehse) [reply examples/flagging-bad-clients/hivemq/Dockerfile:4]: Good point. It is not released yet. However, we will merge once released. ETA very soon :-)

Comment (sfrehse) [reply examples/flagging-bad-clients/hivemq/Dockerfile]: Now uses the released PostgreSQL extension.

File: examples/flagging-bad-clients/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+The solution incorporates the [UserProperties.add](https://docs.hivemq.com/hivemq/4.16/data-governance-hub/policies.html#user-properties-add-function)
+and [Delivery.redirectTo](https://docs.hivemq.com/hivemq/4.16/data-governance-hub/policies.html#delivery-redirect-to-function)
+functions to add MQTT user properties and redirect invalid messages.
+To each redirected message we add user properties to identify bad clients and to analyze the failure.
+
+## Requirements
+- [Docker](https://www.docker.com/) 
+- [Docker compose](https://docs.docker.com/compose/) 
+- A HiveMQ license with enabled Data Governance Hub (contact [datagovernancehub@hivemq.com](mailto:datagovernancehub@hivemq.com))
+- A [HiveMQ Enterprise Extension for PostgreSQL](https://www.hivemq.com/extension/postgresql-extension/) license. If no license is provided, a trial license is valid for 5 hours
```
Comment (gitseti) [inline examples/flagging-bad-clients/README.md:14]:
```suggestion
- A [HiveMQ Enterprise Extension for PostgreSQL](https://www.hivemq.com/extension/postgresql-extension/) license. If no license is provided, a trial license is valid for 5 hours.
```

Comment (gitseti) [reply examples/flagging-bad-clients/README.md:14]: Should be noted where the license has to be placed. Probably as optional point in Quickstart.

File: examples/flagging-bad-clients/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+and [Delivery.redirectTo](https://docs.hivemq.com/hivemq/4.16/data-governance-hub/policies.html#delivery-redirect-to-function)
+functions to add MQTT user properties and redirect invalid messages.
+To each redirected message we add user properties to identify bad clients and to analyze the failure.
+
+## Requirements
+- [Docker](https://www.docker.com/) 
+- [Docker compose](https://docs.docker.com/compose/) 
+- A HiveMQ license with enabled Data Governance Hub (contact [datagovernancehub@hivemq.com](mailto:datagovernancehub@hivemq.com))
+- A [HiveMQ Enterprise Extension for PostgreSQL](https://www.hivemq.com/extension/postgresql-extension/) license. If no license is provided, a trial license is valid for 5 hours.
+## Quickstart
```
Comment (gitseti) [inline examples/flagging-bad-clients/README.md:15]:
```suggestion
- A [HiveMQ Enterprise Extension for PostgreSQL](https://www.hivemq.com/extension/postgresql-extension/) license. If no license is provided, a trial license is valid for 5 hours.

## Quickstart
```

File: examples/flagging-bad-clients/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+The solution incorporates the [UserProperties.add](https://docs.hivemq.com/hivemq/4.16/data-governance-hub/policies.html#user-properties-add-function)
+and [Delivery.redirectTo](https://docs.hivemq.com/hivemq/4.16/data-governance-hub/policies.html#delivery-redirect-to-function)
+functions to add MQTT user properties and redirect invalid messages.
+To each redirected message we add user properties to identify bad clients and to analyze the failure.
+
+## Requirements
+- [Docker](https://www.docker.com/) 
+- [Docker compose](https://docs.docker.com/compose/) 
+- A HiveMQ license with enabled Data Governance Hub (contact [datagovernancehub@hivemq.com](mailto:datagovernancehub@hivemq.com))
+- A [HiveMQ Enterprise Extension for PostgreSQL](https://www.hivemq.com/extension/postgresql-extension/) license. If no license is provided, a trial license is valid for 5 hours.
```
Comment (gitseti) [inline examples/flagging-bad-clients/README.md:14]: Should be noted where the license has to be placed. Probably as optional point in Quickstart.

File: examples/flagging-bad-clients/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+## Quickstart
+
+1. Place your HiveMQ license with enabled Data Governance Hub under `hivemq/hivemq.lic`
+2. Execute `docker compose up`
+3. Execute `mqtt sub -t 'invalid/#' -J` to see redirected MQTT messages from bad clients
```
Comment (gitseti) [inline examples/flagging-bad-clients/README.md:19]: Should we also mention the grafana dashboard?

File: examples/flagging-bad-clients/hivemq/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,20 @@
+FROM hivemq/hivemq4:4.16.0
+
+ARG POSTGRES_EXTENSION_VERSION=1.0.0
+
+# postgres extension
+RUN curl -o /opt/hivemq/extensions/hivemq-postgresql-extension-${POSTGRES_EXTENSION_VERSION#}.zip https://releases.hivemq.com/extensions/hivemq-postgresql-extension-${POSTGRES_EXTENSION_VERSION#}.zip
```
Comment (gitseti) [inline examples/flagging-bad-clients/hivemq/Dockerfile:6]:
```suggestion
RUN curl -o /opt/hivemq/extensions/hivemq-postgresql-extension-${POSTGRES_EXTENSION_VERSION}.zip https://releases.hivemq.com/extensions/hivemq-postgresql-extension-${POSTGRES_EXTENSION_VERSION}.zip
```

**Pull request merged**